src/ext/ipfsutils.js

```diff
@@ -0,0 +1,152 @@
+import { initAutoIPFS } from "@webrecorder/wabac/src/ipfs";
+import { Downloader } from "../downloader";
+
+import * as UnixFS from "@ipld/unixfs";
+import { CarWriter } from "@ipld/car";
+
+export async function ipfsPinUnpin(collLoader, collId, isPin, progress = null) {
+  const coll = await collLoader.loadColl(collId);
+  if (!coll) {
+    return {error: "collection_not_found"};
+  }
+
+  // eslint-disable-next-line no-undef
+  const autoipfs = await initAutoIPFS({web3StorageToken: __WEB3_STORAGE_TOKEN__});
+
+  if (isPin) {
+    const filename = "webarchive.wacz";
+
+    const dl = new Downloader({coll, filename, gzip: false});
+    const dlResponse = await dl.download(progress);
+
+    if (!coll.config.metadata.ipfsPins) {
+      coll.config.metadata.ipfsPins = [];
+    }
+
+    const swContent = await fetchBuffer("sw.js");
```

Comment:
Would it make sense to cache these? Might not matter too much since they're pretty small compared to how big archives could get.